opnode/node/node.go

```diff
@@ -1,18 +1,237 @@
 package node
 
-import "context"
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/ethereum-optimism/optimistic-specs/opnode/eth"
+
+	"github.com/ethereum-optimism/optimistic-specs/opnode/l1"
+	"github.com/ethereum-optimism/optimistic-specs/opnode/l2"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/event"
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/ethereum/go-ethereum/ethclient"
+
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+type GenesisConf struct {
+	L2Hash common.Hash `ask:"--l2-hash" help:"Genesis block hash of L2"`
+	L1Hash common.Hash `ask:"--l1-hash" help:"Block hash of L1 after (not incl.) which L1 starts deriving blocks"`
+	L1Num  uint64      `ask:"--l1-num" help:"Block number of L1 matching the l1-hash"`
+}
+
+func (conf *GenesisConf) GetGenesis() l2.Genesis {
+	return l2.Genesis{
+		L1: eth.BlockID{Hash: conf.L1Hash, Number: conf.L1Num},
+		// TODO: if we start from a squashed snapshot we might have a non-zero L2 genesis number
+		L2: eth.BlockID{Hash: conf.L2Hash, Number: 0},
+	}
+}
 
 type OpNodeCmd struct {
+	L1NodeAddrs   []string `ask:"--l1" help:"Addresses of L1 User JSON-RPC endpoints to use (eth namespace required)"`
+	L2EngineAddrs []string `ask:"--l2" help:"Addresses of L2 Engine JSON-RPC endpoints to use (engine and eth namespace required)"`
+
+	LogCmd `ask:".log" help:"Log configuration"`
+
+	Genesis GenesisConf `ask:".genesis" help:"Genesis anchor point"`
+
+	// during later sequencer rollup implementation:
+	// TODO: multi-addrs option (static peers)
+	// TODO: bootnodes option (bootstrap discovery of more peers)
+
+	log log.Logger
+
+	// (combined) source to fetch data from
+	l1Source eth.L1Source
+
+	// engines to keep synced
+	l2Engines []*l2.EngineDriver
+
+	l1Downloader l1.Downloader
+
+	ctx   context.Context
+	close chan chan error
+}
+
+func (c *OpNodeCmd) Default() {
+	c.L1NodeAddrs = []string{"http://127.0.0.1:8545"}
+	c.L2EngineAddrs = []string{"http://127.0.0.1:8551"}
```

Comment:
Not sure what's standardized yet, but you used port `9000` in the rollup node experiment repo
```suggestion
	c.L2EngineAddrs = []string{"http://127.0.0.1:9000"}
```

Reply:
Good catch, but either works I think. 8551 is the standard engine-API port. It really depends where the engine is activated or not to avoid conflicts (L1 might also have the engine port open). Maybe we should just specify a standard port for Optimism engine API usage. Leaving this to the default port for now (assuming we're testing on non-merged L1 testnets)